treescript/src/treescript/github/client.py

```diff
@@ -0,0 +1,138 @@
+            message (str): The commit message to use.
+            additions (Dict): Files to add or update in the commit. Of the form
+                `{<path>: <contents>}` (optional).
+            deletions (List): Files to delete in the commit. Of the form
+                `[<path>]` (optional).
+        """
+        changes = defaultdict(list)
+        if additions:
+            for name, contents in additions.items():
+                changes["additions"].append({"path": name, "contents": base64.b64encode(contents.encode("utf-8")).decode("utf-8")})
+
+        if deletions:
+            changes["deletions"] = [{"path": p} for p in deletions]
+
+        if not changes:
+            log.warn("No changes to commit, aborting.")
+            return
+
+        query = Template(
+            """
+            query getLatestCommit {
+              repository(owner: "$owner", name: "$repo") {
+                object(expression: "$branch") {
+                  oid
+                }
+              }
+            }
+            """
+        )
+        query = query.substitute(owner=self.owner, repo=self.repo, branch=branch)
+        head_oid = (await self._client.execute(query))["repository"]["object"]["oid"]
+
+        query = """
+            mutation ($input: CreateCommitOnBranchInput!) {
+              createCommitOnBranch(input: $input)  {
+                commit { url }
+              }
+            }
+        """
+        variables = {
+            "input": {
+                "branch": {
+                    "repositoryNameWithOwner": f"{self.owner}/{self.repo}",
+                    "branchName": branch,
+                },
+                "message": {"headline": message},
+                "fileChanges": changes,
+                "expectedHeadOid": head_oid,
+            }
+        }
+        await self._client.execute(query, variables=variables)
+
+    async def get_files(self, files: Union[str, List[str]], branch: Optional[str] = None) -> Dict[str, str]:
+        """Get the contents of the specified files.
+
+        Args:
+            files (List): The list of files to retrieve.
+            branch (str): The branch to retrieve the files from. Uses the
+                repository's default branch if unspecified.
+
+        Returns:
+            Dict: The dictionary of file contents of the form `{<path>: <contents>}`.
+        """
+        branch = branch or "HEAD"
+
+        if isinstance(files, str):
+            files = [files]
+
+        # Periods are not legal GraphQL key names.
+        sentinel_dot = "__dot__"
+        query = Template(
+            dedent(
+                """
+            query getFileContents { 
+              repository(owner: "$owner", name: "$repo") {
+                  $fields
+              }
+            }
+            """
+            )
+        )
+        field = Template(
+            dedent(
+                """
+            $name: object(expression: "$branch:$f") {
```

Comment:
Nit: I'm not sure if `$f` is the shorthand for file or field here. I think it's worth spelling that whole word out.